x-pack/filebeat/input/azureeventhub/input.go

```diff
@@ -135,35 +192,51 @@ func (a *azureInput) Wait() {
 }
 
 func (a *azureInput) processEvents(event *eventhub.Event, partitionID string) bool {
-	timestamp := time.Now()
+	processingStartTime := time.Now()
 	azure := mapstr.M{
 		// partitionID is only mapped in the non-eph option which is not available yet, this field will be temporary unavailable
 		//"partition_id":   partitionID,
 		"eventhub":       a.config.EventHubName,
 		"consumer_group": a.config.ConsumerGroup,
 	}
-	messages := a.parseMultipleMessages(event.Data)
-	for _, msg := range messages {
+
+	// update the input metrics
+	a.metrics.receivedMessages.Inc()
+	a.metrics.receivedBytes.Add(uint64(len(event.Data)))
+
+	records := a.parseMultipleRecords(event.Data)
+
+	for _, record := range records {
 		_, _ = azure.Put("offset", event.SystemProperties.Offset)
 		_, _ = azure.Put("sequence_number", event.SystemProperties.SequenceNumber)
 		_, _ = azure.Put("enqueued_time", event.SystemProperties.EnqueuedTime)
 		ok := a.outlet.OnEvent(beat.Event{
-			Timestamp: timestamp,
+			// this is the default value for the @timestamp field; usually the ingest
+			// pipeline replaces it with a value in the payload.
+			Timestamp: processingStartTime,
 			Fields: mapstr.M{
-				"message": msg,
+				"message": record,
 				"azure":   azure,
 			},
 			Private: event.Data,
 		})
 		if !ok {
+			// a.metrics.publishingErrors.Inc()
```

Comment:
IIRC this only every returns false (e.g. `!ok`) upon shutdown to indicate that it's stopping. No other error states are conveyed this way. So I think you are correct here. We should just remove the commented line.

I think as a future improvement to the input metrics we should find a way to expose metrics from the pipeline client(s) associated to the input. This way we can associate the input with counters for drops and duplicates.

Reply:
> I think as a future improvement to the input metrics we should find a way to expose metrics from the pipeline client(s) associated to the input. This way we can associate the input with counters for drops and duplicates.

Yep, I'll look into this in one of the next iterations. I have plans to migrate to input v2 and the new Event Hub SDK.